src/main/java/org/datadog/jenkins/plugins/datadog/listeners/DatadogBuildListener.java

```diff
@@ -334,6 +334,21 @@ public void onDeleted(Run run) {
                 return;
             }
 
+            // If the build already has a result, this could be a Jenkins cleanup operation
```

Comment:
I'm not knowledgeable enough regarding how those deletion happen and how to identify but it seems that we we try to achieve here is to prevent submitting an "AbortedEvent" for builds that are cleaned up.

A second approach could be to check the endTime of the job, if its "old enough" we can consider that this is just a clean up. But that approach could still submit some AbortedEvent if the clean up happens "too fast".

Not saying this approach is better but wanted to make sure we choose the best approach. Your comment mentions that "it could be a Jenkins cleanup operation", we'd rather be confident regarding that assumption before merging. Any thoughts?

Reply:
Agree that we want to get this right.  The potential issue with time gating the results is it will vary wildly depending on how a user has their Jenkins config setup.  They could have a limit of 10 builds for something that runs once a day (builds deleted after 10 days), or they could have a recurring job that flushes things within an hour or two.  Picking a good default would be the hard part, and may end up causing confusion if some events are reported as Aborted and others are silently dropped.

The thought (possibly flawed) for this approach was that a build which had a status already assigned would have sent an event prior to this one, so its progress would have been recorded at Datadog.  Builds which get deleted before reaching that stage seem like the important ones to track.

We could also add an option on whether to send these events and default it to True - it may be that some customers like that "end-to-end" visibility and some would rather clean it up.

Reply:
Have you validated that approach with a test instance of Jenkins? If so I think this is the best approach to choose and we can move forward with it.

Reply:
Yes, have confirmed it on a local test instance and verified that completed jobs didn't have Aborted events sent when they were deleted.

Next will add a test to this PR to make sure we have the functionality covered.  It's a little tricky since the activity spans two existing test classes (`BuildAbortedEventTest.java` and `DatadogBuildListenerTest.java`).